lib/neo4j/active_node/persistence.rb

```diff
@@ -128,17 +128,14 @@ def create!(*args)
         end
       end
 
-      def merge(attributes)
-        neo4j_session.query.merge(n: {self.mapped_label_names => attributes})
-          .on_create_set(n: on_create_props(attributes))
-          .on_match_set(n: on_match_props)
+      def merge(match_attributes, update_attributes = {})
+        find_or_create_query(match_attributes, update_attributes)
+          .on_match_set(n: on_match_props(update_attributes))
```

Comment:
This does that, right? the `match_attributes` are passed down in the same manner as `find_or_create` does, unless I'm missing something? Have an example use-case?